### PR TITLE
SQL: fall back to using the field name for column label

### LINKED ISSUE #38831: SQL: JDBC meta-data is always using an empty string for column label
[The spec](https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getColumnLabel-int-) defines the column label as
> The suggested title is usually specified by the SQL `AS` clause.  If a SQL `AS` is not specified, the value returned from `getColumnLabel` will be the same as the value returned by the `getColumnName` method.

But in our case, it's always returning an empty string.

### PR DESCRIPTION
**Pull request by astefan**
This PR adds a simple fall back mechanism for the column label defined in [ResultSetMetaData](https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getColumnLabel-int-) so that when asked for a column label we always return the field name. 
This PR does not provide the complete implementation of the column label definition; this will be addressed in a follow up issue since it involves more changes internally in ES-SQL (we do not keep around both the original value and the alias of a field so we can use both in JDBC).

This should fix issues with libraries (Ruby's Sequel used in the Logstash `jdbc` input plugin) that use the column label to discover the names to be used in the results.

Fixes https://github.com/elastic/elasticsearch/issues/38831.

### CODE REVIEW

File: x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSetMetaDataTests.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import static org.elasticsearch.xpack.sql.client.StringUtils.EMPTY;
+
+public class JdbcResultSetMetaDataTests extends ESTestCase {
+    
+    private final List<JdbcColumnInfo> columns = Arrays.asList(
+                new JdbcColumnInfo("test_keyword", EsType.KEYWORD, EMPTY, EMPTY, EMPTY, EMPTY, 0),
+                new JdbcColumnInfo("test_integer", EsType.INTEGER, EMPTY, EMPTY, EMPTY, EMPTY, 11),
+                new JdbcColumnInfo("test_double", EsType.DOUBLE, EMPTY, EMPTY, EMPTY, EMPTY, 25)
+            );
```
Comment (matriv) [inline x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSetMetaDataTests.java]: Please add an example like `new JdbcColumnInfo("test_long", EsType.DOUBLE, EMPTY, EMPTY, EMPTY, "long", 25)` to make clear that the label overrides the name.

**Review by matriv — APPROVED** [review]
LGTM. Left a comment.

File: x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSetMetaDataTests.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSetMetaDataTests.java]: You're right. Extended the test. Thanks.

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by costin — APPROVED** [review]
LGTM

**Pull request merged**